src/adapters/collateralized_simple_interest_loan_adapter.ts

```diff
@@ -0,0 +1,110 @@
+import * as Web3 from "web3";
+import { ContractsAPI } from "src/apis";
+import { Assertions } from "src/invariants";
+import { BigNumber } from "utils/bignumber";
+import * as singleLineString from "single-line-string";
+
+const MAX_COLLATERAL_TOKEN_INDEX_HEX = TermsContractParameters.generateHexValueOfLength(39);
+const MAX_COLLATERAL_AMOUNT_HEX = TermsContractParameters.generateHexValueOfLength(62);
+const MAX_GRACE_PERIOD_IN_DAYS_HEX = TermsContractParameters.generateHexValueOfLength(64);
+
+export interface CollateralizedTermsContractParameters {
+    collateralTokenIndex: BigNumber;
+    collateralAmount: BigNumber;
+    gracePeriodInDays: BigNumber;
+}
+
+export namespace TermsContractParameters {
```

Comment:
Feel like this should be in a separate file that is reused among different adapters.

Reply:
That was my plan. Glad we're seeing eye-to-eye.

Reply:
👓